WordPress/Classes/Utility/OptimizelyHelper.swift

```diff
@@ -29,8 +29,22 @@ internal var optimizelyEnableNewSigninFlowKey: OptimizelyVariableKey = Optimizel
         Optimizely.disableSwizzle() // Disable's the Optimizely visual editor.
         Optimizely.sharedInstance().disableGesture = true
         preregisterOptimizelyKeys()
-        Optimizely.startOptimizelyWithAPIToken(ApiCredentials.optimizelyAPIKey(), launchOptions: launchOptions)
-        Optimizely.refreshExperiments()
+
+        // Check for a signed in account, or at least one blog added to the app.
+        // Load Optimizely asych in these cases as we won't be showing the sign-in
+        // screen right away.
+        let blogService = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        blogService.blogCountForAllAccounts()
+        let accountService = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        if accountService.defaultWordPressComAccount() != nil || blogService.blogCountForAllAccounts() > 0 {
```

Comment:
Since this is launch code, it might be worth it to call `blogCountForAllAccounts` before `defaultWordPressComAccount` as I'd bet `blogCountForAllAccounts` is a bit less expensive than `defaultWordPressComAccount`.